fvm/environment/uuids.go

```diff
@@ -195,6 +204,8 @@ func (generator *uUIDGenerator) GenerateUUID() (uint64, error) {
 	}
 
 	// Since the partition counter only goes up to MaxUint56, we can use the
-	// upper 8 bits to represent which partition was used.
-	return (uint64(generator.partition) << 56) | value, nil
+	// 6th byte to represent which partition was used.
+	// (C7 C6) | P | (C5 C4 C3 C2 C1)
+	return ((value & 0xFF_FF00_0000_0000) << 8) | (uint64(generator.partition) << 40) | (value & 0xFF_FFFF_FFFF), nil
```

Comment:
Maybe we could rename `value` into `counter` in order to avoid confusion, since the comment mentions `counter(C)`. 

It would be better if we could rename `getUint64` and `setUint56` into `getCounter` and `setCounter`
```suggestion
	// assemble a UUID value with the partition (P) and the counter (C).
	// Note: partition (P) is represented by the 6th byte
	// (C7 C6) | P | (C5 C4 C3 C2 C1)
	return ((counter & 0xFF_FF00_0000_0000) << 8) | (uint64(generator.partition) << 40) | (counter & 0xFF_FFFF_FFFF), nil```

Reply:
yeah this is great idea, I will rename them now.